go/chat/unfurl/utils.go

```diff
@@ -0,0 +1,48 @@
+package unfurl
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+
+	"github.com/keybase/client/go/protocol/chat1"
+	"golang.org/x/net/publicsuffix"
+)
+
+func GetHostname(uri string) (res string, err error) {
+	parsed, err := url.Parse(uri)
+	if err != nil {
+		return res, err
+	}
+	return parsed.Hostname(), nil
+}
+
+func GetDomain(uri string) (res string, err error) {
+	parsed, err := url.Parse(uri)
+	if err != nil {
+		return res, err
+	}
+	hostname := parsed.Hostname()
+	if len(hostname) == 0 {
+		return res, errors.New("no hostname")
+	}
+	return publicsuffix.EffectiveTLDPlusOne(hostname)
```

Comment:
if`publicsuffix.EffectiveTLDPlusOne("foo.bar.golang.org") -> "golang.org"` why not split that on `.` and just return `golang`? would make a `switch` in `ClassifyDomain` possible and we can drop the `IsDomain` function